### PR TITLE
[xdoctest] reformat example code with google style in No.21-30

### PR DESCRIPTION
**Pull request by ooooo-create**
### PR types

Others

### PR changes

Others

### Description

修改如下文件的示例代码，使其通过 `xdoctest` 检查：

- python/paddle/autograd/py_layer.py
- python/paddle/autograd/saved_tensors_hooks.py
- python/paddle/framework/dtype.py
- python/paddle/framework/framework.py
- python/paddle/framework/io.py
- python/paddle/framework/io_utils.py
- python/paddle/framework/random.py
- python/paddle/nn/initializer/Bilinear.py
- python/paddle/nn/initializer/assign.py
- python/paddle/nn/initializer/constant.py

预览：

- en:http://preview-paddle-pr-55849.paddle-docs-preview.paddlepaddle.org.cn/documentation/docs/en/api/paddle/autograd/PyLayer_en.html
- zh: http://preview-paddle-pr-55849.paddle-docs-preview.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/autograd/PyLayer_cn.html

关联 PR： 
https://github.com/PaddlePaddle/Paddle/issues/55629
https://github.com/PaddlePaddle/Paddle/pull/55295

@sunzhongkai588 @SigureMo @megemini

### CODE REVIEW

File: python/paddle/autograd/py_layer.py [hunk 1]
```diff
@@ -290,30 +291,31 @@ class PyLayer(with_mateclass(PyLayerMeta, core.eager.PyLayer, PyLayerContext)):
...
+            ...         grad = dy * (1 - paddle.square(y))
+            ...         return grad
+
+            >>> paddle.seed(2023)
+            >>> data = paddle.randn([2, 3], dtype="float64")
+            >>> data.stop_gradient = False
+            >>> z = cus_tanh.apply(data)
+            >>> z.mean().backward()
+
+            >>> print(data.grad)
```
Comment (megemini) [inline python/paddle/autograd/py_layer.py:318]: 如果有输出的话后面加上～

``` python
>>> print(data.grad)
Tensor(shape=[2, 3], dtype=float64, place=Place(cpu), stop_gradient=True,
[[0.16604150, 0.05858341, 0.14051214],
[0.15677770, 0.01564609, 0.02991660]])
```

File: python/paddle/framework/io.py [hunk 2]
```diff
@@ -938,115 +939,116 @@ def load(path, **configs):
...
-            load_main = paddle.load(path)
-            print(load_main)
+            >>> data = paddle.static.data(
+            ...     name='x_static_save', shape=(None, 224), dtype='float32')
+            >>> y_static = z = paddle.static.nn.fc(data, 10)
+            >>> main_program = paddle.static.default_main_program()
+            >>> path = "example/main_program.pdmodel"
+            >>> paddle.save(main_program, path)
+            >>> load_main = paddle.load(path)
+            >>> print(load_main)
```
Comment (megemini) [inline python/paddle/framework/io.py]: 这里感觉没有 print 的必要～ 去掉？

File: python/paddle/nn/initializer/Bilinear.py [hunk 3]
```diff
@@ -36,29 +36,30 @@ class Bilinear(Initializer):
...
-            B = 8
-            H = W = 32
-            w_attr = paddle.ParamAttr(learning_rate=0.,
+            >>> factor = 2
+            >>> C = 2
+            >>> B = 8
+            >>> H = W = 32
+            >>> w_attr = paddle.ParamAttr(learning_rate=0.,
                                       regularizer=L2Decay(0.),
                                       initializer=nn.initializer.Bilinear())
```
Comment (megemini) [inline python/paddle/nn/initializer/Bilinear.py]: 这两行加上 `... ` 这个前缀吧～

不加前缀 xdoctest 也能兼容，但指不定哪天不用 xdoctest 了，这就出问题了～

另外，这个文件不是用 convert-doctest 转换的？按理说应该可以自动加上前缀的，bug ? ... ...

File: python/paddle/nn/initializer/constant.py [hunk 4]
```diff
@@ -88,18 +88,19 @@ class Constant(ConstantInitializer):
...
+            >>> paddle.seed(2023)
+            >>> data = paddle.rand([30, 10, 2], dtype='float32')
+            >>> linear = nn.Linear(2,
+            ...                     4,
+            ...                     weight_attr=nn.initializer.Constant(value=2.0))
+            >>> res = linear(data)
+            >>> print(linear.weight)
+            Tensor(shape=[2, 4], dtype=float32, place=Place(cpu), stop_gradient=False,
+            [[2., 2., 2., 2.],
+             [2., 2., 2., 2.]])
```
Comment (megemini) [inline python/paddle/nn/initializer/constant.py]: 少了一行

``` python
Parameter containing:
Tensor(shape=[2, 4], dtype=float32, place=Place(cpu), stop_gradient=False,
[[2., 2., 2., 2.],
[2., 2., 2., 2.]])
```

File: python/paddle/nn/initializer/Bilinear.py (hunk 3 above)
Comment (ooooo-create) [reply python/paddle/nn/initializer/Bilinear.py]: 这里没有使用convert-doctest，纯属打漏了

File: python/paddle/autograd/py_layer.py (hunk 1 above)
Comment (ooooo-create) [reply python/paddle/autograd/py_layer.py:318]: @megemini 函数里有print,调用函数时会打印，这种需要写输出吗，python/paddle/autograd/saved_tensors_hooks.py中
`       >>> # Example1
        >>> import paddle

        >>> def pack_hook(x):
        ...     print("Packing", x)
        ...     return x.numpy()

        >>> def unpack_hook(x):
        ...     print("UnPacking", x)
        ...     return paddle.to_tensor(x)

        >>> a = paddle.ones([3,3])
        >>> b = paddle.ones([3,3]) * 2
        >>> a.stop_gradient = False
        >>> b.stop_gradient = False
        >>> with paddle.autograd.saved_tensors_hooks(pack_hook, unpack_hook):
        ...     y = paddle.multiply(a, b)
        >>> y.sum().backward()`

Comment (ooooo-create) [reply python/paddle/autograd/py_layer.py:318]: > 如果有输出的话后面加上～
> 
> ```python
> >>> print(data.grad)
> Tensor(shape=[2, 3], dtype=float64, place=Place(cpu), stop_gradient=True,
> [[0.16604150, 0.05858341, 0.14051214],
> [0.15677770, 0.01564609, 0.02991660]])
> ```

函数里有print,调用函数时会打印，这种需要写输出吗，python/paddle/autograd/saved_tensors_hooks.py中
`       >>> # Example1
        >>> import paddle

        >>> def pack_hook(x):
        ...     print("Packing", x)
        ...     return x.numpy()

        >>> def unpack_hook(x):
        ...     print("UnPacking", x)
        ...     return paddle.to_tensor(x)

        >>> a = paddle.ones([3,3])
        >>> b = paddle.ones([3,3]) * 2
        >>> a.stop_gradient = False
        >>> b.stop_gradient = False
        >>> with paddle.autograd.saved_tensors_hooks(pack_hook, unpack_hook):
        ...     y = paddle.multiply(a, b)
        >>> y.sum().backward()`

Comment (megemini) [reply python/paddle/autograd/py_layer.py:318]: 我觉得优先以原示例为准吧～ 如果原来有，就把新的输出改上去，如果没有就算了～

File: python/paddle/framework/random.py [hunk 5]
```diff
@@ -75,8 +75,9 @@ def get_rng_state(device=None):
         GeneratorState:  object.
     Examples:
         .. code-block:: python
-            import paddle
-            sts = paddle.get_rng_state()
+            >>> import paddle
+            >>> paddle.seed(2023)
```
Comment (SigureMo) [inline python/paddle/framework/random.py]: 这里也是有必要 seed 的么？是要固定什么的随机性呢？

File: python/paddle/framework/random.py [hunk 6]
```diff
@@ -129,8 +130,9 @@ def get_cuda_rng_state():
     Examples:
         .. code-block:: python
 
-            import paddle
-            sts = paddle.get_cuda_rng_state()
+            >>> import paddle
+            >>> paddle.seed(2023)
```
Comment (SigureMo) [inline python/paddle/framework/random.py]: 同上

File: python/paddle/framework/random.py [hunk 7]
```diff
@@ -158,9 +160,10 @@ def set_rng_state(state_list, device=None):
     Examples:
         .. code-block:: python
 
-            import paddle
-            sts = paddle.get_rng_state()
-            paddle.set_rng_state(sts)
+            >>> import paddle
+            >>> paddle.seed(2023)
```
Comment (SigureMo) [inline python/paddle/framework/random.py]: 同上

File: python/paddle/framework/random.py [hunk 8]
```diff
@@ -223,9 +226,10 @@ def set_cuda_rng_state(state_list):
     Examples:
         .. code-block:: python
 
-            import paddle
-            sts = paddle.get_cuda_rng_state()
-            paddle.set_cuda_rng_state(sts)
+            >>> import paddle
+            >>> paddle.seed(2023)
```
Comment (SigureMo) [inline python/paddle/framework/random.py]: 同上

File: python/paddle/nn/initializer/Bilinear.py [hunk 9]
```diff
@@ -36,29 +36,30 @@ class Bilinear(Initializer):
...
+            >>> from paddle.regularizer import L2Decay
+
+            >>> factor = 2
+            >>> C = 2
+            >>> B = 8
+            >>> H = W = 32
+            >>> w_attr = paddle.ParamAttr(learning_rate=0.,
+            ...                           regularizer=L2Decay(0.),
+            ...                           initializer=nn.initializer.Bilinear())
+            >>> paddle.seed(2023)
```
Comment (SigureMo) [inline python/paddle/nn/initializer/Bilinear.py]: 这里没输出也不需要固定 seed？

**Review by SigureMo — APPROVED** [review]
LGTMeow～

**Comment by luotao1** [comment]
hi, @ooooo-create 
- 非常感谢你对飞桨框架的贡献，我们正在运营一个PFCC组织，会通过定期分享技术知识与发布开发者主导任务的形式持续为飞桨框架做贡献，详情可见 https://github.com/luotao1 主页说明。
- 如果你对PFCC有兴趣，请发送邮件至 [ext_paddle_oss@baidu.com](mailto:ext_paddle_oss@baidu.com)，我们会邀请你加入~
